File: c#-20260820T054914Z-1-001/c#/Van.cs
// Inheritance
// Van class inherits properties like vehicleId, vehicleNumber,
// and capacity from the Vehicle class.

public class Van : Vehicle
{
    // Van class has its own property as loadCapacity
    private double loadCapacity;

    public Van(int vehicleId, string vehicleNumber, int capacity, double loadCapacity)
        : base(vehicleId, vehicleNumber, capacity)
    {
        this.loadCapacity = loadCapacity;
    }

    // Method overriding
    // Displays van details
    public override void displayVehicleDetails()
    {
        Console.WriteLine("\n----------- Van Details -----------");
        Console.WriteLine("Vehicle ID : " + vehicleId);
        Console.WriteLine("Vehicle Number : " + vehicleNumber);
        Console.WriteLine("Capacity : " + capacity);
        Console.WriteLine("Load Capacity : " + loadCapacity + " kg");
    }
}

File: c++/Van.cpp
#include <iostream>
#include "Van.h"

using namespace std;

Van::Van(int vehicleId, string vehicleNumber, int capacity, double loadCapacity)
    : Vehicle(vehicleId, vehicleNumber, capacity)
{
    this->loadCapacity = loadCapacity;
}

// Displays van details
void Van::displayVehicleDetails()
{
    cout << "\n----------- Van Details -----------" << endl;
    cout << "Vehicle ID : " << vehicleId << endl;
    cout << "Vehicle Number : " << vehicleNumber << endl;
    cout << "Capacity : " << capacity << endl;
    cout << "Load Capacity : " << loadCapacity << " kg" << endl;
}